pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/impl/FixedByteSingleColumnMultiValueReaderWriter.java

```diff
@@ -142,28 +176,42 @@ public void close() {
       dataBuffer.close();
     }
     dataBuffers.clear();
-    headerBuffer.close();
+    for (PinotDataBuffer headerBuffer : headerBuffers) {
```

Comment:
we don't want to call close on the buffer reader/writer since they use the same buffer that we use. Otherwise, we will get double-close.